server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala

```diff
@@ -0,0 +1,618 @@
+        |						"textSignature": "Some text signature",
+        |						"htmlSignature": "<p>Some html signature</p>"
+        |					},
+        |					"4f29": {
+        |						"name": "Identity2",
+        |						"email": "bob@domain.tld",
+        |						"replyTo": null,
+        |						"bcc": null,
+        |						"textSignature": "Some text signature",
+        |						"htmlSignature": "<p>Some html signature</p>"
+        |					}
+        |				}
+        |			},
+        |			"c1"
+        |		]
+        |	]
+        |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1]")
+      .isEqualTo(
+        s""" {
+           | 	"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           | 	"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           | 	"created": {
+           | 		"4f28": {
+           |
+           | 		},
+           | 		"4f29": {
+           |
+           | 		}
+           | 	}
+           | }""".stripMargin)
+  }
+
+  @Test
+  def setIdentityShouldReturnForbiddenFromErrorWhenForbiddenEmailProperty(): Unit = {
+    val request: String =
+      """{
+        |	"using": ["urn:ietf:params:jmap:core"],
+        |	"methodCalls": [
+        |		[
+        |			"Identity/set",
+        |			{
+        |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+        |				"create": {
+        |					"4f28": {
+        |						"name": "valid send from identity",
+        |						"email": "bob@domain.tld"
+        |					},
+        |					"4f29": {
+        |						"name": "forbidden send from identity",
+        |						"email": "bob-alias@domain.tld"
+        |					}
+        |				}
+        |			},
+        |			"c1"
+        |		]
+        |	]
+        |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1]")
+      .isEqualTo(
+        s""" {
+           |	"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |	"newState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
+           |	"created": {
+           |		"4f28": {
+           |			"textSignature": "",
+           |			"htmlSignature": ""
+           |		}
+           |	},
+           |	"notCreated": [
+           |		[
+           |			"4f29",
+           |			{
+           |				"type": "forbiddenFrom",
+           |				"description": "Can not send from bob-alias@domain.tld"
+           |			}
+           |		]
+           |	]
```

Comment:
This is not a correct format for a map!

```suggestion
           |	"notCreated": {
           |         "4f29" : {
           |		"type": "forbiddenFrom",
           |		"description": "Can not send from bob-alias@domain.tld"
           |	     }
           |    }
```